pkg/kubelet/kubelet.go

```diff
@@ -2535,12 +2584,21 @@ func (kl *Kubelet) HandlePodUpdates(pods []*v1.Pod) {
 	start := kl.clock.Now()
 	for _, pod := range pods {
 		kl.podManager.UpdatePod(pod)
-		if kubetypes.IsMirrorPod(pod) {
-			kl.handleMirrorPod(pod, start)
-			continue
+
+		pod, mirrorPod, wasMirror := kl.podManager.GetPodAndMirrorPod(pod)
+		if wasMirror {
+			if pod == nil {
+				klog.V(2).InfoS("Unable to find pod for mirror pod, skipping", "mirrorPod", klog.KObj(mirrorPod), "mirrorPodUID", mirrorPod.UID)
```

Comment:
If the log lines included the pod.UID it would enable better debugging if the mirror pod can't be found. We could include this in a followup PR.

Reply:
+1, my gap

Reply:
Ok, this is slightly tricky.  If we get called here we don't know the pod UID, because `pod` was a mirror pod, and mirror pods don't have the source UID on them (that I know of).  Definitely a followup.

Reply:
Yeah, we don't have any way to find the real UID, because all we have is:

```
metadata:
  annotations:
    kubernetes.io/config.hash: 063e897b129c1f6d595e9010a8ac70ea // the hash
    kubernetes.io/config.mirror: 063e897b129c1f6d595e9010a8ac70ea // the hash
    kubernetes.io/config.seen: "2023-05-08T17:32:16.730248098Z"
    kubernetes.io/config.source: file
```